src/FunctionTimer.cpp

```diff
@@ -0,0 +1,199 @@
+///////////////////////////////////////////////////////////////////////////////
+///
+///	\file    FunctionTimer.cpp
+///	\author  Paul Ullrich
+///	\version July 26, 2010
+///
+///	<remarks>
+///		Copyright 2021 Paul Ullrich
+///
+///		This file is distributed as part of the Tempest source code package.
+///		Permission is granted to use, copy, modify and distribute this
+///		source code and its documentation under the terms of the GNU General
+///		Public License.  This software is provided "as is" without express
+///		or implied warranty.
+///	</remarks>
+
+#include "FunctionTimer.h"
+#include "Exception.h"
+
+#include <iostream>
+#include <sys/time.h>
+
+///////////////////////////////////////////////////////////////////////////////
+
+FunctionTimer::GroupDataMap FunctionTimer::m_mapGroupData;
+
+///////////////////////////////////////////////////////////////////////////////
+
+FunctionTimer::FunctionTimer(const char *szGroup) {
+
+	// Start the timer
+	m_fStopped = false;
+
+	// Assign group name
+	if (szGroup == NULL) {
+		m_strGroup = "";
+	} else {
+		m_strGroup = szGroup;
+	}
+
+	// Assign start time
+	gettimeofday(&m_tvStartTime, NULL);
```

Comment:
It is more portable to use the `high_resolution_clock` provided with standard C++11.

https://en.cppreference.com/w/cpp/chrono/high_resolution_clock

Reply:
You can still have your FunctionTimer storing context and details on top - but sys/time.h is tricky to include portably.